extractor/text_utils.go

```diff
@@ -0,0 +1,249 @@
+/*
+ * This file is subject to the terms and conditions defined in
+ * file 'LICENSE.md', which is part of this source code package.
+ */
+
+package extractor
+
+import (
+	"math"
+	"sort"
+	"unicode"
+)
+
+// serial is used to add serial numbers to all text* instances.
+var serial serialState
```

Comment:
dangerous for concurrent use.  Example: using multiple go routines to extract from different documents

Reply:
I removed it.